test/RestrictedNORI-transfer.test.ts

```diff
@@ -0,0 +1,531 @@
+import { BigNumber } from 'ethers';
+
+import {
+  expect,
+  advanceTime,
+  batchMintAndListRemovalsForSale,
+} from '@/test/helpers';
+import {
+  setupTestLocal,
+  restrictRemovalProceeds,
+  compareScheduleDetailForAddressStructs,
+  compareScheduleSummaryStructs,
+  SECONDS_IN_10_YEARS,
+} from '@/test/helpers/restricted-nori';
+
+describe('RestrictedNORI transferring', () => {
```

Comment:
it's unclear to me why this needs to be in a separate file. _**if**_ it does need to be, then I would export this as a function and then call it in the other file. separating the tests like this makes it easier to duplicate a tests intention in separate files

Reply:
the reason I split them out originally was that the linter was so laggy in the large file that things were hard to work on. i'll recombine and we can split them out if this becomes a problem for anyone in the future

Reply:
maybe when we overhaul the fixtures and create some better abstractions for setup we'll be able to reduce a bunch of boilerplate and make this test file a lot shorter too, which will help with the lint lag for those of us who aren't running 2056 cores or whatever

Reply:
lol definitely. I'll try to take a pass on this after it's merged, and it would be easier to do so if I had everything in one file. FWIW if the linter/formatter is getting slowed down in a file of < 2000 lines, there's probably some other issue with the linter itself and it's trying to evaluate more rules than it needs to, i'll follow up with a quick investigation

Reply:
just ran an analysis on my laptop, and while a  ~1 second is slow-ish, I'm not sure it's worth the tradeoff of splitting it into separate files for such a case. I ran the same against a file with 100 lines, and it was only 200ms faster, so not sure we even achieve much in terms of speed by splitting it. Maybe you made some additional changes that caused this to speed up? Any idea why it was running slow for you previously. Maybe it had nothing to do with the linter or formatter and had to do with some other issue I'm not thinking of?

```
TIMING=1 npx eslint @nori-dot-com/contracts/test/RestrictedNORI.test.ts
Rule                       | Time (ms) | Relative
:--------------------------|----------:|--------:
prettier/prettier          |   926.374 |    67.2%
import/namespace           |    44.722 |     3.2%
jest/no-identical-title    |    24.512 |     1.8%
jest/no-conditional-expect |    23.764 |     1.7%
jest/no-standalone-expect  |    22.755 |     1.7%
jest/no-done-callback      |    22.457 |     1.6%
jest/expect-expect         |    21.299 |     1.5%
jest/no-test-prefixes      |    18.187 |     1.3%
jest/valid-title           |    18.180 |     1.3%
jest/no-focused-tests      |    17.813 |     1.3%
Rule | Time (ms) | Relative
:----|----------:|--------:

```

Reply:
hahah thank you for being such a scientist nerd.  it does seem possible to me that something else was going on with VSCode or my machine that was making things so unwieldy because working in this big test file now doesn't feel so painful.  will leave combined.

Reply:
:+1: it's actually possible it's not the rule evaluation that is the issue. I dug deeper on this after posting and realized if you run the above test with --debug, you can see the time overhead for all the steps including setup before evaluation.  The main overhead seems to be coming from the fact that typescript-eslint compiles before linting, which can add a couple seconds in a large file. Worth monitoring

https://typescript-eslint.io/docs/linting/type-linting